File: backend/models/Activity.js
import mongoose from "mongoose";

const activitySchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: String,
  cityId: { type: mongoose.Schema.Types.ObjectId, ref: 'City', required: true },
  category: { type: String, enum: ['sightseeing', 'food', 'adventure', 'culture', 'other'] },
  cost: { type: Number },
  duration: { type: Number }, // in hours
  images: [{ type: String }],
  createdAt: { type: Date, default: Date.now }
});

export default mongoose.model("Activity", activitySchema);

File: python/models/activity.py
from pydantic import BaseModel, Field
from typing import Optional, List

class ActivityCreate(BaseModel):
    name: str
    description: Optional[str] = None
    category: Optional[str] = Field(None, description="One of sightseeing, food, adventure, culture, other")
    cost: Optional[float] = None
    duration: Optional[float] = None  # in hours

class ActivitiesResponseModel(BaseModel):
    activities: List[ActivityCreate]
